### ISSUE TITLE
MA0183 False-positive

### ISSUE THREAD

**Issue by dmitriy-maksimov-wd** [meziantou/Meziantou.Analyzer#989]
### Version of the Meziantou.Analyzer NuGet package

2.0.292

### Rule Identifier

MA0183

### Target Framework

Net10.0

### C# Language version

C# 14

### Description

Getting the following error
```
Error MA0183 : Use string literal instead of string.Format when the format string has no placeholders
```

But placeholder are in the string and have parameters passed to `string.Format()`

### Reproduction Steps

Minimal code:

```c#
class Program
{
	private static string DebuggerDisplay => string.Format(CultureInfo.InvariantCulture, "Column: {0}, Value: {1}, Invalid: {2}, Blank: {3}", "Column", "Value", true, false);

	static void Main(string[] args)
    {
        Console.WriteLine(DebuggerDisplay);
    }
}
```

**Comment by dmitriy-maksimov-wd** [comment]
Another example
```
class Program
{
    static void Main(string[] args)
    {
          Console.WriteLine(string.Format("{0} x {1} [{2} x {3}{4}{5}]", 0, 1, 2, 3, 4, args.Length));
    }
}
```

<img width="1317" height="68" alt="Image" src="https://github.com/user-attachments/assets/731b6321-3c6a-443d-8958-1cb80a097ca9" />

**Comment by Draeggiar** [comment]
Same problem here, in my case it may be because of the verbatim/raw string:
``` c#
string.Format(CultureInfo.InvariantCulture, @"<svg width=""{0}"" height=""{0}"" xmlns=""http://www.w3.org/2000/svg"">
  <circle cx=""50%"" cy=""50%"" r=""50%"" fill=""{3}"" />
  <text x=""50%"" y=""55%"" style=""font-family:'Segoe UI Semibold','SF Pro Display Medium','Segoe UI','Sans','Arial'"" font-size=""{1}"" font-weight=""normal"" fill=""{4}"" text-anchor=""middle"" dominant-baseline=""middle"">{2}</text>
</svg>", sizePx, fontSize, name, color.BackgroundColor, color.FontColor)
```

**Comment by meziantou** [comment]
It should be fixed now with the latest version (2.0.293). This happens because you are using 4 parameters, which uses the overload with `params object[]`. Because the compiler creates an implicit array, the rule didn't work correctly. There were some logics about the implicit array, but it was expected for empty arrays.

**Comment by dmitriy-maksimov-wd** [comment]
The issue still exists in latest `2.0.293` when compiling using .Net 10.0.1
The same code

```
class Program
{
	private static string DebuggerDisplay => string.Format("Zero: {0}, One: {1}, Two: {2}, Four: {3}", "Answer is", 42, true, false);

	static void Main(string[] args)
    {
        Console.WriteLine(DebuggerDisplay);
        Console.WriteLine(string.Format("{0} x {1} [{2} x {3}{4}{5}]", 0, 1, 2, 3, 4, args.Length));
    }
}
```
@meziantou do you want me to raise a new issue (I can't reopen this one)?

**Comment by meziantou** [comment]
The version 2.0.294 seems to fix the issue with .NET 10. I think this is because of new overloads with span and collection expressions.
If you find other false positive, please open new issues.
